back/lib/shared/auth/auth.go

```diff
@@ -37,19 +38,22 @@ func createNewPasswordHash(rawPassword string) (password string, salt string, er
 }
 
 func generateSalt() (string, error) {
-	salt := make([]byte, constants.SaltLength)
-	max := big.NewInt(int64(len(constants.Chars)))
-
-	for i := range salt {
-		randomNumber, err := rand.Int(rand.Reader, max)
+	rawSalt := make([]uint16, constants.SaltLength)
+	for i := range rawSalt {
+		randomNumber, err := rand.Int(rand.Reader, big.NewInt(
+			constants.LastSaltCharacter-constants.FirstSaltCharacter))
 		if err != nil {
 			return "", err
 		}
 
-		salt[i] = constants.Chars[randomNumber.Int64()]
+		sanitizedRandomNumber := randomNumber.Int64() + constants.FirstSaltCharacter
+		rawSalt[i] = uint16(sanitizedRandomNumber)
 	}
 
-	return string(salt), nil
+	runes := utf16.Decode(rawSalt)
+	salt := string(runes)
+
+	return salt, nil
```

Comment:
You can remove the salt declaration just above and send it directly

```suggestion
	return string(runes), nil
```